cimsparql/queries.py

```diff
@@ -555,6 +554,38 @@ def two_winding_transformer_query(
         network_analysis,
         with_loss,
     )
+    where_list.extend(
+        [
+            "?w_mrid_2 cim:PowerTransformerEnd.phaseAngleClock ?angleclock",
+            f"optional {{{tap_phase_angle('?w_mrid_2', '?_angle')}}}",
+            "bind((coalesce (?_angle+xsd:float(30.0)*?angleclock, xsd:float(0.0))) as ?angle)",
+        ]
+    )
+
+    return sup.combine_statements(sup.select_statement(variables), sup.group_query(where_list))
+
+
+def tap_phase_angle(mrid: str = "?mrid", angle: str = "?angle") -> str:
+    variables = [mrid, angle]
+    where_list = [
+        common_subject(
+            "?tap_step",
+            [
+                "rdf:type cim:SvTapStep",
+                "cim:SvTapStep.TapChanger ?tap_changer",
+                "cim:SvTapStep.position ?position",
+            ],
+        ),
+        common_subject(
+            "?tap_changer",
+            [
+                "cim:PhaseTapChangerLinear.stepPhaseShiftIncrement ?inc",
+                "cim:TapChanger.normalStep ?normalstep",
+                f"cim:PhaseTapChanger.TransformerEnd {mrid}",
+            ],
+        ),
+        f"bind(((?position-?normalstep)*xsd:float(str(?inc))) as {angle})",
+    ]
     return sup.combine_statements(sup.select_statement(variables), sup.group_query(where_list))
```

Comment:
Add test that confirms that the calculation is correct (e.g. position-normalstep) and that the resulting value is a `float`. Something like

```python
df = tap_changer()

assert df["angle"] is not NaN
```

Against a fixed model such as the micro model (if that model has linear phase tap changers) one could also check against expected angle numbers

```python
assert df["angle"].iloc[0] == pytest.approx(0.3)  # Or whatever the number is supposed to be
```